### PR TITLE
WisdomWarLogo

### PR DESCRIPTION
**Pull request by khalosmoscato**
This pull request introduces the InputPages.tsx component designed for creating quizzes. The following features have been implemented:

1. Quiz Title Input
A text input field for the quiz title.
Validation to ensure the title is not empty, with an error message displayed if validation fails.
2. Quiz Description Input
A text area for entering a brief quiz description.
Validation to ensure the description is not empty, with an error message displayed if validation fails.
3. Category Dropdown
A dropdown menu for selecting the quiz category.
Validation to ensure a category is selected, with an error message displayed if validation fails.
4. Difficulty Dropdown
A dropdown menu for selecting the quiz difficulty level.
Validation to ensure a difficulty level is selected, with an error message displayed if validation fails.
5. Questions and Answers Section
Allows users to add a minimum of 5 questions, each with:
A text input for the question.
A text input for the correct answer.
Three text inputs for wrong answers.
Validation checks to ensure:
Questions are not empty and are unique.
The correct answer differs from the wrong answers.
All wrong answers are filled and unique.
6. Error Messages
Error messages are displayed in red next to the fields that have issues.
A summary error message appears above the submit buttons if any validation errors exist, prompting users to fix the errors.
7. Buttons
Add More Questions: Allows users to dynamically add additional question blocks.
Submit Quiz: Validates inputs and displays a modal upon successful validation.
Navigation Buttons in Modal:
Back to Home: Navigates to the homepage.
Explore Quizzes: Navigates to the explore quizzes page.
8. Modal
Displays a thank-you message upon successful quiz creation.
Provides navigation options to return to the homepage or explore quizzes.
Summary
The InputPages.tsx component has been thoroughly tested for user experience and validation functionality, ensuring users receive clear feedback when creating quizzes. This implementation improves the overall usability of the quiz creation process.

### CODE REVIEW

**Comment by khalosmoscato** [comment]
I forgot to mention, I also updated the singular components:
- all the input components follow a sanitize and validate process to avoid users to type malicious code and cause issues to the project.
- the NavBar components are now working, by clicking the back arrow, you go back one page, by clicking mute and unmute the icon changes instead of having 2 icons on the screen.
- the buttons have more functionalities 
- I created the logo page, the logo image can be changed in the future, this one is just for reference.
- I reorganised App.tsx and Main.tsx so that they are more lightweight and only contain the code they are supposed to.

**Comment by khalosmoscato** [comment]
If you go to my PLOG of this week I broke down each functionality to explain exactly where they are implemented and how they work, if you want to have a [look:](https://github.com/fac30/PLOG_khalos/blob/main/Week_04.md)

File: wisdom-war/src/components/inputs/CategoryDropdown.tsx [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
-const CategoryDropdown: React.FC = () => {
+interface CategoryDropdownProps {
+  onChange: (value: string) => void; // Prop to handle category selection
+}
+
+const CategoryDropdown: React.FC<CategoryDropdownProps> = ({ onChange }) => {
   const [selectedCategory, setSelectedCategory] = useState<string>('Select Category');
   const [isOpen, setIsOpen] = useState<boolean>(false); // State to manage dropdown visibility
 
   const categories = ['FAC', 'Music', 'Movies', 'Art'];
```
Comment (heyitzi) [inline wisdom-war/src/components/inputs/CategoryDropdown.tsx:12]: This needs to be connected to the backend to retrieve the Categories. We can reuse the code I wrote last week to get all the categories available in the quizzes. 
Also, we would need another option so the users can add more categories.

File: wisdom-war/src/components/NavBar.tsx [hunk 2]
```diff
@@ -1,29 +1,32 @@
...
-            <RiVolumeUpFill />
-          </div>
+    <nav className="navbar flex-col w-full px-4 py-8">
+      <div className="section-navbar flex justify-between items-center px-4 pb-4">
+        <div className="title-section flex justify-start items-center gap-4 ">
+          <RiArrowLeftSLine onClick={() => navigate(-1)} /> {/* Navigate to the previous page on click */}
+          <h2 className="navTitle text-2xl">{props.title}</h2>
+        </div>
+        <div className="audio-section flex items-center" onClick={toggleVolume}>
+          {isMuted ? <RiVolumeMuteFill /> : <RiVolumeUpFill />} {/* Show icon based on mute state */}
```
Comment (heyitzi) [inline wisdom-war/src/components/NavBar.tsx:24]: Are we connecting the music to these buttons?

File: README.md [hunk 3]
```diff
@@ -98,4 +98,318 @@ const App = () => {
...
+     - **Back to Home**: Navigates to the homepage.
+     - **Explore Quizzes**: Navigates to the explore quizzes page.
+
+### 8. **Modal**
+   - Displays a thank-you message upon successful quiz creation.
+   - Contains navigation options to return to the homepage or explore quizzes.
+
+## Code Overview
+
+Here is the code for the `InputPages.tsx` component:
```
Comment (heyitzi) [inline README.md]: Do we need the code for the input pages in the readme?

Comment (khalosmoscato) [reply README.md]: Deleted, you are right. Our code is public and we do not need the code here.

**Review by MarikaBBB — APPROVED** [review]

**Pull request merged**